pkg/yaml/yaml.go

```diff
@@ -146,3 +148,37 @@ func findScalarNode(node *yaml.Node, keyPath []string) (int, int, error) {
 	}
 	return 0, 0, fmt.Errorf("key path not found")
 }
+
+// MergeFiles merges a list of YAML files.
+func MergeFiles(inputPaths []string, outputPath string) error {
+	if len(inputPaths) == 0 || outputPath == "" {
+		return fmt.Errorf("inFiles and OutFile must not be empty")
+	}
+
+	// read first YAML file
+	mergedNode, err := kyaml.ReadFile(inputPaths[0])
+	if err != nil {
+		return fmt.Errorf("error parsing first input file: %w", err)
+	}
+
+	// read all other YAML file and apply the patch
+	for i := 1; i < len(inputPaths); i++ {
+		patchNode, fileErr := kyaml.ReadFile(inputPaths[i])
+		if fileErr != nil {
+			return fmt.Errorf("error parsing input file %s: %w", inputPaths[i], fileErr)
+		}
+
+		mergedNode, err = merge2.Merge(patchNode, mergedNode, kyaml.MergeOptions{ListIncreaseDirection: 1})
+		if err != nil {
+			return fmt.Errorf("error merging file %s: %w", inputPaths[i], err)
+		}
+	}
```

Comment:
I feel maybe this can optimise the logic, 
```suggestion
	// read first YAML file
	mergedNode, err := kyaml.ReadFile(inputPaths[0])
	if err != nil {
		return fmt.Errorf("error parsing first input file: %w", err)
	}

	// read all other YAML files and merge them
	for _, inputPath := range inputPaths[1:] {
		patchNode, err := kyaml.ReadFile(inputPath)
		if err != nil {
			return fmt.Errorf("error parsing input file %s: %w", inputPath, err)
		}

		mergedNode, err = merge2.Merge(patchNode, mergedNode, kyaml.MergeOptions{ListIncreaseDirection: 1})
		if err != nil {
			return fmt.Errorf("error merging file %s: %w", inputPath, err)
		}
	}
```